src/components/learner-credit-management/cards/NewAssignmentModalButton.jsx

```diff
@@ -84,32 +124,63 @@ const NewAssignmentModalButton = ({ course, children }) => {
           httpErrorStatus,
           httpErrorResponseData,
         } = err.customAttributes;
+        let locallyScopedErrorReason = 'system_error';
         if (httpErrorStatus === 422) {
           const responseData = JSON.parse(httpErrorResponseData);
           setCreateAssignmentsErrorReason(responseData[0].reason);
+          locallyScopedErrorReason = responseData[0].reason;
         } else {
           setCreateAssignmentsErrorReason('system_error');
         }
         setAssignButtonState('error');
+        sendEnterpriseTrackEvent(
+          enterpriseId,
+          EVENT_NAMES.LEARNER_CREDIT_MANAGEMENT.ASSIGNMENT_ALLOCATION_ERROR,
+          {
+            allocatedLearners: learnerEmails.length,
+            courseUUID: course.uuid,
+            errorStatus: httpErrorStatus,
+            errorReason: locallyScopedErrorReason,
+          },
+        );
       },
     });
   };
 
   return (
     <>
-      <Button onClick={open}>{children}</Button>
+      <Button onClick={handleOpenAssignmentModal}>{children}</Button>
       <FullscreenModal
         className="bg-light-200 text-left"
         title="Assign this course"
         isOpen={isOpen}
-        onClose={handleCloseAssignmentModal}
+        onClose={() => {
+          handleCloseAssignmentModal();
+          sendEnterpriseTrackEvent(
+            enterpriseId,
+            EVENT_NAMES.LEARNER_CREDIT_MANAGEMENT.CLOSE_ASSIGNMENT_MODAL_EXIT,
```

Comment:
[curious/clarification] Would this result in duplicate events regarding the modal closing? For example, this callback function would fire both `EVENT_NAMES.LEARNER_CREDIT_MANAGEMENT.CLOSE_ASSIGNMENT_MODAL_EXIT` and `EVENT_NAMES.LEARNER_CREDIT_MANAGEMENT.CLOSE_ASSIGNMENT_MODAL`.

Reply:
Resolved in an earlier comment 👍🏽